### DISCUSSION TITLE
Example to set margin for collision detection

### DISCUSSION THREAD

**Discussion by Jiayuan-Gu** [stack-of-tasks/pinocchio#1697] [Q&A] [answered]
I notice that there is an interface`setSecurityMargins`, which is relevant to `hpp-fcl` security margin features. But can you help provide an example of how to set margins for a loaded robot in python binding, especially how to use `security_margin_map`? Thanks. 

<https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/structpinocchio_1_1GeometryData.html#ab360f0fcaaa214ab87508516114bd56f>

**Comment by jmirabel** [comment, marked as answer]
I haven't checked the code but it looks straightforward to me:
It is a matrix whose row and column index correspond to geometry index and whose values correspond to the security margin to be used for the corresponding pair of geometries.
Since the tuples (i,j) and (j,i) represent the same pair of geometries, the matrix should be symmetrical. To avoid having to set both values, you can set only the upper or lower part of the matrix and set the boolean argument accordingly.
